Gems/ROS2/Code/Source/Lidar/LidarRegistrarEditorSystemComponent.cpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "LidarRegistrarEditorSystemComponent.h"
+#include <AzCore/Serialization/SerializeContext.h>
+
+namespace ROS2
+{
+    void LidarRegistrarEditorSystemComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<LidarRegistrarEditorSystemComponent, LidarRegistrarSystemComponent>()->Version(0);
+        }
+    }
+
+    void LidarRegistrarEditorSystemComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+    {
+        BaseSystemComponent::GetProvidedServices(provided);
```

Comment:
This piece of code was generated using a SystemComponent template and since the BaseSystemComponent is an alias for LidarRegistrarSystemComponent I saw no purpose in removing it (I'm not exactly sure why the SystemComponent template uses it in the first place). Throughout the engine many gems seem to either remove or leave it.